src/ray/raylet/lib/python/raylet_extension.cc

```diff
@@ -3,78 +3,73 @@
 
 #include "common_extension.h"
 #include "config_extension.h"
-#include "ray/raylet/local_scheduler_client.h"
+#include "ray/raylet/raylet_client.h"
 
 PyObject *LocalSchedulerError;
 
 // clang-format off
 typedef struct {
   PyObject_HEAD
-  LocalSchedulerConnection *local_scheduler_connection;
-} PyLocalSchedulerClient;
+  RayletClient *raylet_client;
+} PyRayletClient;
 // clang-format on
 
-static int PyLocalSchedulerClient_init(PyLocalSchedulerClient *self, PyObject *args,
-                                       PyObject *kwds) {
+static int PyRayletClient_init(PyRayletClient *self, PyObject *args, PyObject *kwds) {
   char *socket_name;
   UniqueID client_id;
   PyObject *is_worker;
   JobID driver_id;
   if (!PyArg_ParseTuple(args, "sO&OO&", &socket_name, PyStringToUniqueID, &client_id,
                         &is_worker, &PyObjectToUniqueID, &driver_id)) {
-    self->local_scheduler_connection = NULL;
+    self->raylet_client = NULL;
     return -1;
   }
   /* Connect to the local scheduler. */
-  self->local_scheduler_connection = LocalSchedulerConnection_init(
-      socket_name, client_id, static_cast<bool>(PyObject_IsTrue(is_worker)), driver_id,
-      Language::PYTHON);
+  self->raylet_client = new RayletClient(socket_name, client_id,
+                                         static_cast<bool>(PyObject_IsTrue(is_worker)),
+                                         driver_id, Language::PYTHON);
   return 0;
 }
 
-static void PyLocalSchedulerClient_dealloc(PyLocalSchedulerClient *self) {
-  if (self->local_scheduler_connection != NULL) {
-    LocalSchedulerConnection_free(self->local_scheduler_connection);
+static void PyRayletClient_dealloc(PyRayletClient *self) {
+  if (self->raylet_client != NULL) {
+    delete self->raylet_client;
   }
   Py_TYPE(self)->tp_free((PyObject *)self);
 }
 
-static PyObject *PyLocalSchedulerClient_disconnect(PyObject *self) {
-  local_scheduler_disconnect_client(
-      ((PyLocalSchedulerClient *)self)->local_scheduler_connection);
+static PyObject *PyRayletClient_Disconnect(PyRayletClient *self) {
+  auto status = self->raylet_client->Disconnect();
+  RAY_CHECK_OK_PREPEND(status, "[RayletClient] Failed to disconnect.");
```

Comment:
`RAY_CHECK_OK_PREPEND` looks too strong here. If there is a unsuccessful disconnect call, the client don't need to crash.

Reply:
It looks like `RAY_CHECK_OK_PREPEND` are used everywhere. It really too strict to use it. However, I think it is fine in this PR. We can change it later like https://github.com/ray-project/ray/pull/3493 . It is better to make the caller decide whether to catch this exception instead of crashing by itself.

Reply:
Since now every method in RayletClient returns a status, we can change all `RAY_CHECK_OK_PREPEND` to another macro that throws an exception upwards if status is not ok, and let the caller decide what to do. 
One motivation of having this ability is that sometimes an actor may run background threads, it doesn't want to crash immediately when raylet connection is closed. 
I can work on this once this PR is merged.

Reply:
Yes, another PR could be clearer.